paddle/inference/inference.h

```diff
@@ -17,35 +17,31 @@ limitations under the License. */
 #include "paddle/framework/block_desc.h"
 #include "paddle/framework/lod_tensor.h"
 #include "paddle/framework/program_desc.h"
+#include "paddle/framework/executor.h"
+#include "paddle/framework/scope.h"
 
 namespace paddle {
 
 class InferenceEngine {
 public:
-  InferenceEngine() : program_(nullptr), load_program_(nullptr) {}
-  ~InferenceEngine() {
-    delete program_;
-    delete load_program_;
-  }
-
-  void LoadInferenceModel(const std::string& dirname);
-  void LoadInferenceModel(const std::string& dirname,
-                          const std::vector<std::string>& feed_var_names,
-                          const std::vector<std::string>& fetch_var_names);
-  void Execute(const std::vector<framework::LoDTensor>& feeds,
+  InferenceEngine() : program_(nullptr) {}
+  ~InferenceEngine() { delete program_; }
+
+  void LoadInferenceModel(framework::Executor& executor,
+                          framework::Scope& scope,
+                          const std::string& dirname);
```

Comment:
The definition of `InferenceEngine` will be deleted (or you can delete it in this PR). I suggest to implement all load and save functions as a global function in a separate file named `io.cc`, just like `io.py` in Python API.

About the function's name, I suggest we use `Load(...)` directly.

Reply:
Yes, I just kept this to have a minimal InferenceEngine as of now to get the load() working. Just to clarify, this calls the `LoadModelAndParam` which is a new namespace `infer` (in inference.cc). So as you suggested, in the future, i will move this to a separate file. Do you think the name `LoadModelAndParam` is not okay?

Reply:
> Do you think the name LoadModelAndParam is not okay?

I think we can `Load(...)` for simple.

> in the future, i will move this to a separate file

I suggest doing this now to fix the interface for users as soon as possible.

Reply:
Okay will fix naming and file structure. Thanks.